### PR TITLE
fix #113051: Arpeggio playback should not depend on tempo

### PR DESCRIPTION
**Pull request by handrok**
1) Add Stretch field to arpeggio inspector to manipulate velocity of arpeggio. Now arpeggio with Stretch which equal 1, doesn't depend on applying tempo. Arpeggio sounds the same.
2) Add unit test to midi tests.

### CODE REVIEW

File: libmscore/arpeggio.h [hunk 1]
```diff
@@ -79,7 +81,9 @@ class Arpeggio final : public Element {
 
       bool playArpeggio()       { return _playArpeggio; }
       void setPlayArpeggio(bool p) { _playArpeggio = p; }
-
+
```
Comment (mattmcclinch) [inline libmscore/arpeggio.h]: You removed a blank line and added a line with only whitespace. This should be avoided.

File: libmscore/rendermidi.cpp [hunk 2]
```diff
@@ -946,7 +946,11 @@ void renderArpeggio(Chord *chord, QList<NoteEventList> & ell)
       for (int i = start; i != end; i += step) {
             NoteEventList* events = &(ell)[i];
             events->clear();
-            int ot = (l * j * 1000) / chord->upNote()->playTicks();
+            auto tempoRatio = chord->score()->tempomap()->tempo(chord->tick()) / Score::defaultTempo();
+            int ot = (l * j * 1000) / chord->upNote()->playTicks() * 
+                        tempoRatio * chord->arpeggio()->Stretch();
+            
+            //int ot;
```
Comment (mattmcclinch) [inline libmscore/rendermidi.cpp]: Surely you didn't mean to leave `//int ot;` here.

File: mscore/inspector/inspectorArpeggio.cpp [hunk 3]
```diff
@@ -28,7 +28,8 @@ InspectorArpeggio::InspectorArpeggio(QWidget* parent)
       g.setupUi(addWidget());
 
       const std::vector<InspectorItem> iiList = {
-            { Pid::PLAY, 0,     g.playArpeggio, g.resetPlayArpeggio}
+            { Pid::PLAY, 0,     g.playArpeggio, g.resetPlayArpeggio},
+            { Pid::TIME_STRETCH,        0, g.stretch,      g.resetStretch }
```
Comment (mattmcclinch) [inline mscore/inspector/inspectorArpeggio.cpp]: It would have been nice to align the columns in this table.

**Comment by Jojo-Schmitz** [comment]
Not too surprisingly the mtest that deals with the midi rendering of arpeggios failed, see https://travis-ci.org/musescore/MuseScore/jobs/431864140#L5552

**Comment by handrok** [comment]
@Jojo-Schmitz not sure what I did wrong. I added this midi test. I just created mscx with arpeggio and exported it to midi. Then added to mtest. Why it failed I don't know. midi and mscx sounds the same

File: mtest/libmscore/midi/tst_midi.cpp [hunk 4]
```diff
@@ -55,6 +55,7 @@ class TestMidi : public QObject, public MTest
       void midiBendsExport1() { midiExportTestRef("testBends1"); }
       void midiBendsExport2() { midiExportTestRef("testBends2"); }      // Play property test
       void midiPortExport()   { midiExportTestRef("testMidiPort"); }
+      void mediArpeggio()     { midiExportTestRef("arpeggio"); }
```
Comment (Jojo-Schmitz) [inline mtest/libmscore/midi/tst_midi.cpp]: typo, should be `midiArpeggio()`

Comment (handrok) [reply mtest/libmscore/midi/tst_midi.cpp]: thanks

**Comment by handrok** [comment]
Don't know how -ref.mid sounds. I haven't linux OS right now.

**Comment by Jojo-Schmitz** [comment]
You shouldn't need Linux for that

File: mtest/libmscore/midi/tst_midi.cpp (hunk 4 above)
Comment (Jojo-Schmitz) [reply mtest/libmscore/midi/tst_midi.cpp]: And probably also `testArpeggio`, so the files should be named testArpeggio.mscx and testArpeggio-ref.mid

File: libmscore/arpeggio.cpp [hunk 5]
```diff
@@ -434,6 +440,8 @@ QVariant Arpeggio::getProperty(Pid propertyId) const
 bool Arpeggio::setProperty(Pid propertyId, const QVariant& val)
       {
       switch(propertyId) {
+            case Pid::TIME_STRETCH:
+                  setStretch(val.toDouble());
```
Comment (Jojo-Schmitz) [inline libmscore/arpeggio.cpp]: is the "Fall THrough" intended here? Of so better add a `/* FALLTHROUGH */`, to a) document and b) avoid a compiler warning

Comment (Jojo-Schmitz) [reply libmscore/arpeggio.cpp]: Actually I believe this is a bug and a `break;` is missing here. For once a valid compiler warning ;-)

**Pull request merged**